pkgs/by-name/go/gordonflashtool/package.nix

```
@@ -0,0 +1,38 @@
+{ lib, stdenv, fetchFromGitHub, nasm, unixtools }:
+
+let
+  version = "10";
+in
+
+stdenv.mkDerivation {
+  pname = "gordonflashtool";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "marmolak";
+    repo = "GordonFlashTool";
+    rev = "release-10";
+    sha256 = "sha256-/zpw7kVdQeR7QcRsP1+qcu8+hlEQTGwOKClJkwVcBPg=";
+  };
+
+  nativeBuildInputs = [ nasm unixtools.xxd ];
+
+  buildPhase = ''
+    # build the gordon binary
+    make all-boot-code
+  '';
```

Add phase hooks so that the package can be overridden to add `preBuild`, `postBuild` etc.
```suggestion
  buildPhase = ''
    runHook preBuild
    # build the gordon binary
    make all-boot-code
    runHook postBuild
  '';
```